src/js/profile/mobile/widget/Calendar.js

```diff
@@ -169,43 +223,96 @@
 			* @protected
 			*/
 			prototype._onClick = function (event) {
-				var calendarElement = this.element.querySelector(".calendarView");
+				var calendarElement = this.element.querySelector(".calendarView"),
+					self = this;
 
 				if (event.srcElement.className === "calendarArrowRight") {
-					todayMonth = todayMonth + 1;
-					if (todayMonth === 13) {
-						todayMonth = 1;
-						todayYear = todayYear + 1;
-					}
-					prototype._deleteCalendar(calendarElement);
-					today = new Date(todayYear, todayMonth - 1);
-					prototype._buildCalendar(calendarElement);
+					self._moveMonth("right", calendarElement);
 				} else if (event.srcElement.className === "calendarArrowLeft") {
-					todayMonth = todayMonth - 1;
-					if (todayMonth === 0) {
-						todayMonth = 12;
-						todayYear = todayYear - 1;
-					}
-					prototype._deleteCalendar(calendarElement);
-					today = new Date(todayYear, todayMonth - 1);
-					prototype._buildCalendar(calendarElement);
+					self._moveMonth("left", calendarElement);
+				} else if (event.srcElement.className === "calendarArrowLeft nonSelectionArrow") {
+					return;
+				} else if (event.srcElement.className === "prevDisableDay nonSelectionDay" ||
+					event.srcElement.firstChild.className === "prevDisableDay nonSelectionDay") {
+					return;
+				} else if (event.srcElement.className === "prevDisableDay") {
+					self._moveMonth("left", calendarElement);
+					this._selection(event.target, event.target.innerHTML);
+				} else if (event.srcElement.className === "nextDisableDay") {
+					self._moveMonth("right", calendarElement);
+					this._selection(event.target, event.target.innerHTML);
 				} else {
 					this._selection(event.target);
 				}
 			};
 
+			/**
+			* Selecting a date leaves a mark
+			* @method _moveMonth
+			* @param {string} direction
+			* @param {HTMLElement} calendarElement
+			* @member ns.widget.mobile.Calendar
+			* @protected
+			*/
+			prototype._moveMonth = function (direction, calendarElement) {
+				var self = this;
+
+				switch (direction) {
+					case "left" :
+						todayMonth = todayMonth - 1;
+						if (todayMonth === 0) {
+							todayMonth = 12;
+							todayYear = todayYear - 1;
+						}
+						self._deleteCalendar(calendarElement);
```

Comment:
DRY (Don't Repeat Yourself).

These line of codes are duplicated below
```
						self._deleteCalendar(calendarElement);
						dateData = new Date(todayYear, todayMonth - 1);
						self._buildCalendar(calendarElement);
```
You can move this code out of conditional.

Reply:
done.